src/scenario/scenario_events_parameter_data.c

```diff
@@ -226,6 +233,67 @@ scenario_action_data_t *scenario_events_parameter_data_get_actions_xml_attribute
     return &scenario_action_data[type];
 }
 
+typedef struct {
+    int type;
+    translation_key key;
+} sorting_attr_t;
+
+static scenario_condition_data_t *scenario_condition_data_alphabetical[CONDITION_TYPE_MAX - 1];
+static scenario_action_data_t *scenario_action_data_alphabetical[ACTION_TYPE_MAX - 1];
+
+static int platform_aware_string_compare(const char *a, const char *b)
+{
+#ifdef _MSC_VER
+    return _mbsicmp((const unsigned char *) a, (const unsigned char *) b);
+#else
+    return strcasecmp(a, b);
+#endif
+}
```

Comment:
Instead of this, you should create your own `string_compare` that uses `tolower` to be case-insensitive.

That also avoids the "funny" stuff of casting the `uint8_t`'s to `char`s.